### ISSUE TITLE
Allow user to specify eslint path

### ISSUE THREAD

**Issue by MikeRatcliffe** [AtomLinter/linter-eslint#584]
It would be great if users could specify their local eslint path e.g. Mozilla's Firefox source now installs a local version of eslint in testing/eslint/node_modules/.bin/eslint and you cannot point to this binary at the moment.

**Comment by MikeRatcliffe** [comment]
The accompanying Mozilla bug is at https://bugzil.la/1273496

**Comment by MikeRatcliffe** [comment]
It is fairly common for each module of a large project to have it's own node_modules folder and this allows people to cherry-pick the folder they want to use.

**Comment by IanVS** [comment]
Continuing discussion from #585.

@MikeRatcliffe is ESLint installed into the `node_modules` of each of your project's modules?  If so, it _should_ work correctly as-is.

Maybe you can explain a little more about the problem you're having, or the error you're getting?  A minimal reproducible test case is the best.

**Comment by MikeRatcliffe** [comment]
**Scenario**

I am in the root of a project with hundreds of modules in various languages. A version of eslint is installed in 1 of them at `testing/eslint/node_modules/` along with a bunch of custom eslint plugins. These node modules must be installed locally and cannot be at the root of the project. This eslint module is used by editors when editing any JS code in the tree.

Any code that fails linting is automatically rejected.

The project is loaded in Atom with linting enabled. How would Atom know which ESLint should be used when linting?

In sublime text you can set paths to the eslint binary:

```
"paths": {
            "linux": [],
            "osx": [
                "/Users/username/Desktop/firefox/testing/eslint/node_modules/.bin/"
            ],
            "windows": []
        },
```

If you want a simple test case this is simple as you can get (Linux or OSX only... Windows is much more complex):

1. hg clone http://hg.mozilla.org/integration/fx-team
2. cd fx-team
3. ./mach eslint --setup
4. After eslint is set up you will be given a path to the eslint binary, something like `path/to/project/testing/eslint/node_modules/.bin/eslint`.
5. Load the project up in Atom with linter-eslint enabled.
6. Ensure that "Don't use .eslintignore files" is checked.
6. Open `devtools/client/inspector/inspector-panel.js`.

**Expected**
A bunch of eslint warnings should be displayed.

**Actual**
No warnings are displayed because it is not possible to have Atom point at the correct version of eslint.

There are a number of Mozilla devs that have started using Atom and it would be a shame if they had to stop using the editor simply because they are unable to choose their own eslint path.

**Comment by Arcanemagus** [comment]
So what you are describing above is a _direct_ contradiction of what you said [here](https://github.com/AtomLinter/linter-eslint/pull/585#issuecomment-220300686), which was that each module needs it's own version of `eslint` (which is what is supported currently).

Is everything supposed to be using a single unified version of `eslint`, or does each module have it's own version that it is verified to work with?

**Comment by MikeRatcliffe** [comment]
@Arcanemagus Sorry for the confusion. What I meant to say was that each module needs to be able to choose the eslint package from another module. Sorry for the confusion.

I will add some specs and we can go from there.

**Comment by Arcanemagus** [comment]
Fixed in #609.
